lighthouse-core/gather/gather-runner.js

```diff
@@ -172,6 +173,22 @@ class GatherRunner {
     }
   }
 
+  /**
+   * Returns an error if the original network request failed or wasn't found.
+   * @param {LH.Crdp.Security.SecurityStateChangedEvent} securityState
+   * @return {LHError|undefined}
+   */
+  static checkForSecurityIssue({securityState, explanations}) {
```

Comment:
bikeshed: if it does throw in here, maybe rename `assertCertificateError` or something? Or does this catch a larger class of issues

Reply:
It catches anytime the latest security state has an insecure property. a certificate error is one example. I don't know if it's just one of, most of them, or all possible scenarios.

Reply:
Just triple checking because the protocol name is getting me concerned a bit, but `securityState === 'insecure'` only when the page is HTTPS but actually isn't secure, right? Not just every time the page is served over HTTP without encryption correct? :)

Reply:
+1 to the idea of throwing in here with a `assert*` rename too, btw. Seems much easier and removes the cognitive load from the caller

Reply:
Good idea to check HTTP. Tested with `node lighthouse-cli http://www.httpvshttps.com/ --verbose --view` - it still generated a full LHR. I presume "insecure" is only used for HTTPS connection that don't comply with security standards (and I guess they always go hand-in-hand with interstitial security warnings).